docs/admin/kubelet-tls-bootstrapping.md

```diff
@@ -0,0 +1,96 @@
+is in place, scoping the tokens to a group will allow great flexibility (e.g. you could disable a particular
+bootstrap group's access when you are done provisioning the nodes). 
+
+### Token auth file 
+Tokens are arbitrary but should represent at least 128 bits of entropy derived from a secure random number 
+generator (such as /dev/urandom on most modern systems). There are multiple ways you can generate a token. For example: 
+
+`head -c 16 /dev/urandom | od -An -t x | tr -d ' '` 
+
+will generate tokens that look like `02b50b05283e98dd0fd71db496ef01e8` 
+
+The token file will look like the following example, where the first three values can be anything and the quoted group 
+name should be as depicted: 
+
+```
+02b50b05283e98dd0fd71db496ef01e8,kubelet-bootstrap,10001,"system:kubelet-bootstrap"
+``` 
+
+Add the `--token-auth-file=FILENAME` flag to the apiserver command to enable the token file. 
+See docs at http://kubernetes.io/docs/admin/authentication/#static-token-file for further details.
+
+### Client certificate CA bundle
+
+Add the `--client-ca-file=FILENAME` flag to the apiserver command to enable client certificate authentication,
+referencing a certificate authority bundle containing the signing certificate.
+
+## controller-manager configuration
+The API for requesting certificates adds a certificate-issuing control loop to the KCM. This takes the form of a 
+[cfssl](https://blog.cloudflare.com/introducing-cfssl/) local signer using assets on disk. 
+Currently, all certificates issued have one year validity and a default set of key usages. 
+
+### Signing assets 
+You must provide a Certificate Authority in order to provide the cryptographic materials necessary to issue certificates. 
+This CA should be trusted by the apiserver for authentication with the `--client-ca-file=SOMEFILE` flag. The management
+of the CA is beyond the scope of this document but it is recommended that you generate a dedicated CA for Kubernetes.
+Both certificate and key are assumed to be PEM-encoded. 
+
+The new controller-manager flags are: 
+```
+--cluster-signing-cert-file="/etc/path/to/kubernetes/ca/ca.crt" --cluster-signing-key-file="/etc/path/to/kubernetes/ca/ca.key"
+``` 
+
+### Auto-approval 
+To ease deployment and testing, the alpha version of the certificate request API includes a flag to approve all certificate 
+requests made by users in a certain group. The intended use of this is to whitelist only the group corresponding to the bootstrap
+token in the token file above. Use of this flag circumvents makes the "approval" process described below and is not recommended
+for production use. 
+
+The flag is: 
+```
+--insecure-experimental-approve-all-kubelet-csrs-for-group="system:kubelet-bootstrap"
+``` 
+
+## kubelet configuration 
+To use request a client cert from the certificate request API, the kubelet needs a path to a kubeconfig file that contains the
```

Comment:
this was a straight move of the content in `kubelet-tls-bootstrapping.md`, I'd rather save content changes in that file for a follow-up